<?xml version="1.0" encoding="utf-8" ?>
<launch>
    <arg name="robot" default="spot-3"/><group if="$(eval {'ctu-robot': 0, 'flip': 0, 'marv-robot': 0, 'spot-3': 0}[robot])"/>

    <arg name="camera_topic" value="/viz/camera_4/image/compressed" if="$(eval robot == 'ctu-robot')"/>
    <arg name="camera_topic" value="TODO" if="$(eval robot == 'flip')"/>
    <arg name="camera_topic" value="/camera_front/image_color/compressed" if="$(eval robot == 'marv-robot')"/>
    <arg name="camera_topic" value="/camera_front/image_color/compressed" if="$(eval robot == 'spot-3')"/>

    <arg name="segmentation_topic" value="/viz/camera_4/image_segmentation/compressed" if="$(eval robot == 'ctu-robot')"/>
    <arg name="segmentation_topic" value="TODO" if="$(eval robot == 'flip')"/>
    <arg name="segmentation_topic" value="/camera_front/image_segmentation/compressed" if="$(eval robot == 'marv-robot')"/>
    <arg name="segmentation_topic" value="/camera_front/image_segmentation/compressed" if="$(eval robot == 'spot-3')"/>

    <node name="segmentation_node" pkg="image_segmentation" type="segmentation.py" output="screen">

        <!-- input -->
        <remap from="/image_to_segment/compressed" to="$(arg camera_topic)"/>

        <!-- output -->
        <remap from="/segmented_image/compressed" to="$(arg segmentation_topic)"/>

    </node>
</launch>
